stacks-signer/src/v0/signer.rs

```diff
@@ -0,0 +1,318 @@
+                "{self}: received a block proposal for a new block. Submit block for validation. ";
+                "signer_sighash" => %signer_signature_hash,
+            );
+            let block_info = BlockInfo::from(block_proposal.clone());
+            stacks_client
+                .submit_block_for_validation(block_info.block.clone())
+                .unwrap_or_else(|e| {
+                    warn!("{self}: Failed to submit block for validation: {e:?}",);
+                });
+            self.signer_db
+                .insert_block(&block_info)
+                .unwrap_or_else(|_| panic!("{self}: Failed to insert block in DB"));
+            return;
+        };
+        if let Some(block_response) = self.determine_response(&mut block_info) {
+            // Submit a proposal response to the .signers contract for miners
+            warn!("{self}: Broadcasting a block response to stacks node: {block_response:?}");
+            if let Err(e) = self
+                .stackerdb
+                .send_message_with_retry(block_response.into())
+            {
+                warn!("{self}: Failed to send block rejection to stacker-db: {e:?}",);
+            }
+        }
+        self.signer_db
+            .insert_block(&block_info)
+            .unwrap_or_else(|_| panic!("{self}: Failed to insert block in DB"));
+    }
+
+    /// Handle the block validate response returned from our prior calls to submit a block for validation
+    fn handle_block_validate_response(&mut self, block_validate_response: &BlockValidateResponse) {
+        debug!("{self}: Received a block validate response: {block_validate_response:?}");
+        let (message, block_info) = match block_validate_response {
+            BlockValidateResponse::Ok(block_validate_ok) => {
+                crate::monitoring::increment_block_validation_responses(true);
+                let signer_signature_hash = block_validate_ok.signer_signature_hash;
+                // For mutability reasons, we need to take the block_info out of the map and add it back after processing
+                let mut block_info = match self
+                    .signer_db
+                    .block_lookup(self.reward_cycle, &signer_signature_hash)
+                {
+                    Ok(Some(block_info)) => block_info,
+                    Ok(None) => {
+                        // We have not seen this block before. Why are we getting a response for it?
+                        debug!("{self}: Received a block validate response for a block we have not seen before. Ignoring...");
+                        return;
+                    }
+                    Err(e) => {
+                        error!("{self}: Failed to lookup block in signer db: {e:?}",);
+                        return;
+                    }
+                };
+                block_info.valid = Some(true);
+                let signature = self
+                    .private_key
+                    .sign(&signer_signature_hash.0)
```

Comment:
This is pretty nit but we mainly use `.bits()` in other places (though there is almost certainly both used with Sha512Trunc256Sum in other places)